classport-agent/src/test/java/ClassportAgentTest.java

```diff
@@ -0,0 +1,164 @@
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import org.junit.jupiter.api.Test;
+
+import io.github.chains_project.classport.agent.ClassportAgent;
+import io.github.chains_project.classport.commons.AnnotationReader;
+import io.github.chains_project.classport.commons.ClassportInfo;
+
+public class ClassportAgentTest {
+
+    private static final String RESOURCES_PATH = "src/test/resources/";
+    private static final String CLASSPORT_TREE_PATH = RESOURCES_PATH + "classport-deps/classport-deps-tree";
+    private static final String CLASSPORT_LIST_PATH = RESOURCES_PATH + "classport-deps/classport-deps-list";
+    private static final String ANNOTATED_CLASS_PATH = RESOURCES_PATH + "annotated-classes/StringUtils.class";
+    private static final String NOT_ANNOTATED_CLASS_PATH = RESOURCES_PATH + "not-annotated-classes/StringUtils.class";
+
+    @Test
+    void shouldGenerateDependencyListAndTreeFiles() throws Exception {
+        HashMap<String, ClassportInfo> sbom = getSBOM();
+        
+        // Invokation of private method WriteSBOM of ClassportAgent class
+        invokeWriteSBOMMethod(sbom);
+
+        File treeFile = new File(System.getProperty("user.dir"), "classport-deps-tree");
+        File listFile = new File(System.getProperty("user.dir"), "classport-deps-list");
+        
+        assertAll(
+            () -> assertTrue(treeFile.exists(), "Tree output file should be created"),
+            () -> assertTrue(listFile.exists(), "List output file should be created"),
+            () -> assertTrue(compareFiles(treeFile, new File(CLASSPORT_TREE_PATH)), "Tree files should be identical."),
+            () -> assertTrue(compareFiles(listFile, new File(CLASSPORT_LIST_PATH)), "List files should be identical.")
+        );
+
+        treeFile.delete();
+        listFile.delete();
+    }
+
+    
+    @Test
+    void shouldHandleEmptySBOM() throws Exception {
+        HashMap<String, ClassportInfo> emptySBOM = new HashMap<>();
```

Comment:
We want to assert that [`getSBOM`](https://github.com/chains-project/classport/blob/7df77a244d74341d7392a3a29ee834cd23191610/classport-analyser/src/main/java/io/github/chains_project/classport/analyser/Analyser.java#L33) can handle empty SBOM so if you create empty map here, it does not test for the actual method `getSBOM` that will run in production.